### DISCUSSION TITLE
Is NodeJS pull mode working?

### DISCUSSION THREAD

**Discussion by brunobastosg** [grafana/pyroscope#2477] [Q&A]
Hi, all!

According to the [NodeJS Language SDK documentation](https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/nodejs/), to use pull mode, I need to add [scrape configuration](https://grafana.com/docs/pyroscope/latest/configure-client/language-sdks/nodejs/#scrape-configuration) to the Pyroscope config file.

However, in the [configuration reference](https://grafana.com/docs/pyroscope/latest/configure-server/reference-configuration-parameters/), there is no mention of a configuration parameter called `scrape-configs`.

I even checked the source code, but couldn't find anything.

Even so, I started Pyroscope with a custom config file with the `scrape-configs` block as instructed. There was no error, but no scraping was being done.

I also checked the [express-pull](https://github.com/grafana/pyroscope/tree/main/examples/nodejs/express-pull) example in the repository, but it uses the Grafana Agent instead of a `scrape-configs` block in the configuration file.

So, is the documentation wrong? Or is this an upcoming feature? Or should I use the Grafana Agent?

Thank you in advance!

**Comment by cyriltovena** [comment]
You're right the documention is wrong, we removed the scraping ability directly from Pyroscope since this was a duplication of effort with Grafana Agent.

We wanted to integrate the agent back into Pyroscope but this hasn't been a priority.

So if you want pull mode you'll need to install Grafana Agent in flow mode with profiling scrape_config.
